metals/src/main/scala/scala/meta/internal/builds/Digest.scala

```diff
@@ -0,0 +1,251 @@
+package scala.meta.internal.builds
+
+import java.nio.charset.StandardCharsets
+import java.nio.file.Files
+import java.nio.file.Path
+import java.security.MessageDigest
+import java.util.stream.Collectors
+
+import scala.meta.internal.builds.Digest.Status
+import scala.meta.internal.io.PathIO
+import scala.meta.internal.mtags.MD5
+import scala.meta.internal.mtags.MtagsEnrichments._
+import scala.meta.io.AbsolutePath
+import scala.meta.tokens.Token
+import scala.util.control.NonFatal
+import scala.collection.JavaConverters._
+
+case class Digest(
+    md5: String,
+    status: Status,
+    millis: Long
+)
+
+object Digest {
+
+  /**
+   * Bump up this version if parameters outside of the sbt sources themselves require
+   * re-running `bloopInstall`. For example a SemanticDB or Bloop version upgrade.
+   */
+  val version: String = "v4"
+  sealed abstract class Status(val value: Int)
+      extends Product
+      with Serializable {
+    import Status._
+    def isRequested: Boolean = this == Requested
+    def isStarted: Boolean = this == Started
+    def isRejected: Boolean = this == Rejected
+    def isFailed: Boolean = this == Failed
+    def isInstalled: Boolean = this == Installed
+    def isCancelled: Boolean = this == Cancelled
+  }
+  object Status {
+    case object Requested extends Status(0)
+    case object Started extends Status(1)
+    case object Rejected extends Status(2)
+    case object Failed extends Status(3)
+    case object Installed extends Status(4)
+    case object Cancelled extends Status(5)
+    case class Unknown(n: Int) extends Status(n)
+    def all: List[Status] = List(
+      Requested,
+      Started,
+      Rejected,
+      Failed,
+      Installed,
+      Cancelled
+    )
+  }
+
+  def digestDirectory(
+      path: AbsolutePath,
+      digest: MessageDigest
+  ): Boolean = {
+    if (!path.isDirectory) true
+    else {
+      var isSuccess = true
+      Files.list(path.toNIO).forEach { file =>
+        isSuccess = isSuccess && digestFile(AbsolutePath(file), digest)
+      }
+      isSuccess
+    }
+  }
+
+  def digestFileBytes(
+      path: AbsolutePath,
+      digest: MessageDigest
+  ): Boolean = {
+    if (path.isFile) {
+      digest.update(path.readAllBytes)
+    }
+    true
+  }
+
+  def digestFile(
+      path: AbsolutePath,
+      digest: MessageDigest
+  ): Boolean = {
+    val ext = PathIO.extension(path.toNIO)
+    val isScala = ext match {
```

Comment:
would `ext == "sbt" || ext == "scala"` be more readable?
or even `Set("scala", "sbt").contains(ext)` -  since there already is a precedence of using sets with `isGradle` ;)